src/plain/number/arithmetic.js

```diff
@@ -345,3 +345,22 @@ export function normNumber (x) {
   return Math.abs(x)
 }
 normNumber.signature = n1
+
+/**
+ * Calculates the modular multiplicative inverse of a mod b
+ * @param {number} a
+ * @param {number} b
+ * @return {number|null}
+ */
+export function modMultiplicativeInverseNumber (a, b) {
+  // source: https://en.wikipedia.org/wiki/Modular_multiplicative_inverse#Extended_Euclidean_algorithm
+  if (b === 0) {
+    throw new Error('divisor must be an integer')
+  }
+  const [div, m] = xgcdNumber(a, b)
```

Comment:
@josdejong I think I cannot extend the argument type to both float and big number since `xgcdNumber` throws an error if the params are non-integer. I need to update this function's logic in a way it can handle non-integer as well.

Reply:
@ohpyupi The `xgcd` function indeed only works with integers.

To get a working solution for `BigNumber`, you can inject `xgcd` instead of using `xgcdNumber`, this works for number and BigNumber. Like: 
```js
const dependencies = ['typed', 'xgcd']
// ...
export const createModMultiplicativeInverse = /* #__PURE__ */ factory(name, dependencies, ({ typed, xgcd }) => {
  // ...
  const [div, m] = xgcd(a, b)
  // ...
}
```

Reply:
@josdejong Oh cool! But what about the float type? Does it work with the float type as well?

Reply:
The function `xgcd` does not support floats, only integer values. Function `mod` does support floats. If needed for `modMultiplicativeInverseNumber ` I guess we have to extend `xgcd`?

Reply:
What about this? In this scope of PR, I ignore the float support for the modular inverse. And if the float support is requested, it can be implemented then. let me know if this approach works on your end.

Reply:
Yes that sounds like a pragmatic approach :+1: